torch/_inductor/pattern_matcher.py

```diff
@@ -1839,11 +1844,22 @@ def placeholder(
         def call_function(
             self, target: str, args: Sequence[Any], kwargs: Mapping[str, Any]  # type: ignore[override]
         ) -> PatternExpr:
-            args, kwargs = pytree.tree_map(process_arg, (args, kwargs))
+            # Indexing is critical for matching getitem nodes, so we can't ignore int args here
+            if target == operator.getitem and int in ignore_types:
+                temp_ignore_types = tuple(t for t in ignore_types if t is not int)
+
+                def process_arg_fun(
+                    x: T, ignore_types_override: Optional[Sequence[Type[Any]]] = None
+                ) -> Union[T, KeywordArg, Ignored]:
+                    return process_arg(x, ignore_types_override=temp_ignore_types)
+
+            else:
+                process_arg_fun = process_arg
```

Comment:
nit: if you name `process_arg_fn` above `process_arg` no need for else clause

Reply:
fixed in https://github.com/pytorch/pytorch/pull/140193/commits/0da92a4f1daf9c4f2a7221884b5beb528e4f1c66